### ISSUE TITLE
java.lang.NullPointerException initializing embedded servlet

### ISSUE THREAD

**Issue by irinacen** [eclipse/jetty.project#4064]
Embedded servlet raises a NullPointerException when servlet is called on version 9.4.20.v20190813. 

Same issue with JDK8 and JDK11.

This DOESN'T happen in any of the previous versions I've tried, including inmediate previous version 9.4.19.v20190610, with the exactly same code, configuration and environment.

The issue can be reproduced running the "Minimal Servlet" sample from official documentation: https://www.eclipse.org/jetty/documentation/current/embedded-examples.html#embedded-minimal-servlet

When GET call is done to listening ip and port, the received response is HTTP ERROR 500 and the servlet error trace is:

```
2019-09-06 09:35:55.347:INFO::main: Logging initialized @616ms to org.eclipse.jetty.util.log.StdErrLog
2019-09-06 09:35:55.521:INFO:oejs.Server:main: jetty-9.4.20.v20190813; built: 2019-08-13T21:28:18.144Z; git: 84700530e645e812b336747464d6fbbf370c9a20; jvm 1.8.0_202-b08
2019-09-06 09:35:56.358:INFO:oejs.AbstractConnector:main: Started ServerConnector@dcf3e99{HTTP/1.1,[http/1.1]}{0.0.0.0:8080}
2019-09-06 09:35:56.359:INFO:oejs.Server:main: Started @1651ms
2019-09-06 09:36:03.267:INFO:oeju.TypeUtil:qtp1394438858-16: JVM Runtime does not support Modules
2019-09-06 09:36:03.303:WARN:oejsh.ContextHandler:qtp1394438858-16: Unimplemented - use org.eclipse.jetty.servlet.ServletContextHandler
2019-09-06 09:36:03.306:WARN:oejsh.ContextHandler:qtp1394438858-16: unavailable
java.lang.NullPointerException
	at org.eclipse.jetty.servlet.ServletHolder.initServlet(ServletHolder.java:656)
	at org.eclipse.jetty.servlet.ServletHolder.getServlet(ServletHolder.java:511)
	at org.eclipse.jetty.servlet.ServletHolder.prepare(ServletHolder.java:791)
	at org.eclipse.jetty.servlet.ServletHandler.doHandle(ServletHandler.java:539)
	at org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:190)
	at org.eclipse.jetty.servlet.ServletHandler.doScope(ServletHandler.java:482)
	at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:141)
	at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:127)
	at org.eclipse.jetty.server.Server.handle(Server.java:494)
	at org.eclipse.jetty.server.HttpChannel.handle(HttpChannel.java:374)
	at org.eclipse.jetty.server.HttpConnection.onFillable(HttpConnection.java:268)
	at org.eclipse.jetty.io.AbstractConnection$ReadCallback.succeeded(AbstractConnection.java:311)
	at org.eclipse.jetty.io.FillInterest.fillable(FillInterest.java:103)
	at org.eclipse.jetty.io.ChannelEndPoint$2.run(ChannelEndPoint.java:117)
	at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:782)
	at org.eclipse.jetty.util.thread.QueuedThreadPool$Runner.run(QueuedThreadPool.java:918)
	at java.lang.Thread.run(Thread.java:748)
2019-09-06 09:36:03.309:WARN:oejs.HttpChannel:qtp1394438858-16: /
javax.servlet.ServletException: carmelo.embeddedjetty.HelloServlet-2a33fae0@2c5b2ff6==carmelo.embeddedjetty.HelloServlet,jsp=null,order=-1,inst=false,async=true
	at org.eclipse.jetty.servlet.ServletHolder.initServlet(ServletHolder.java:677)
	at org.eclipse.jetty.servlet.ServletHolder.getServlet(ServletHolder.java:511)
	at org.eclipse.jetty.servlet.ServletHolder.prepare(ServletHolder.java:791)
	at org.eclipse.jetty.servlet.ServletHandler.doHandle(ServletHandler.java:539)
	at org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:190)
	at org.eclipse.jetty.servlet.ServletHandler.doScope(ServletHandler.java:482)
	at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:141)
	at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:127)
	at org.eclipse.jetty.server.Server.handle(Server.java:494)
	at org.eclipse.jetty.server.HttpChannel.handle(HttpChannel.java:374)
	at org.eclipse.jetty.server.HttpConnection.onFillable(HttpConnection.java:268)
	at org.eclipse.jetty.io.AbstractConnection$ReadCallback.succeeded(AbstractConnection.java:311)
	at org.eclipse.jetty.io.FillInterest.fillable(FillInterest.java:103)
	at org.eclipse.jetty.io.ChannelEndPoint$2.run(ChannelEndPoint.java:117)
	at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:782)
	at org.eclipse.jetty.util.thread.QueuedThreadPool$Runner.run(QueuedThreadPool.java:918)
	at java.lang.Thread.run(Thread.java:748)
Caused by: 
java.lang.NullPointerException
	at org.eclipse.jetty.servlet.ServletHolder.initServlet(ServletHolder.java:656)
	at org.eclipse.jetty.servlet.ServletHolder.getServlet(ServletHolder.java:511)
	at org.eclipse.jetty.servlet.ServletHolder.prepare(ServletHolder.java:791)
	at org.eclipse.jetty.servlet.ServletHandler.doHandle(ServletHandler.java:539)
	at org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:190)
	at org.eclipse.jetty.servlet.ServletHandler.doScope(ServletHandler.java:482)
	at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:141)
	at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:127)
	at org.eclipse.jetty.server.Server.handle(Server.java:494)
	at org.eclipse.jetty.server.HttpChannel.handle(HttpChannel.java:374)
	at org.eclipse.jetty.server.HttpConnection.onFillable(HttpConnection.java:268)
	at org.eclipse.jetty.io.AbstractConnection$ReadCallback.succeeded(AbstractConnection.java:311)
	at org.eclipse.jetty.io.FillInterest.fillable(FillInterest.java:103)
	at org.eclipse.jetty.io.ChannelEndPoint$2.run(ChannelEndPoint.java:117)
	at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:782)
	at org.eclipse.jetty.util.thread.QueuedThreadPool$Runner.run(QueuedThreadPool.java:918)
	at java.lang.Thread.run(Thread.java:748)
```

**Comment by joakime** [comment]
I've been going through and adding testing to the embedded examples via PR #4056
Haven't reached the `MinimalServlets` example yet.
I'll do that one next.

**Comment by joakime** [comment]
That's a badly named example.
It's really a MinimalServlet (no plural), and it has no ServletContext, doesn't support Filters, and a whole litany of other gotchas.

**Comment by joakime** [comment]
I have replicate this issue.

This change is what broke it - https://github.com/eclipse/jetty.project/commit/819c37997516ca49b1b00978344c5a8c2f600f08#diff-0d419a5cace2d3e164cd481d96f24734

**Comment by joakime** [comment]
Commit 7618eae91534178a375e021a65c141da19833c9d fixes this (and is now part of PR #4056)

**Comment by gregw** [comment]
@joakime It would be good to release 9.4.21 ASAP, so could  7618eae be handled separately from a general cleanup of the embedded code.... or can you complete that cleanup ASAP?

**Comment by joakime** [comment]
Let me make that PR non-draft, so you can review it.
Stay tuned.

**Comment by joakime** [comment]
@gregw PR #4056 is ready for review

**Comment by joakime** [comment]
PR #4056 is now merged into `jetty-9.4.x`.
Expect the fix in the upcoming 9.4.21 release.
